src/software/SfM/main_openMVG2ABC.cpp

```diff
@@ -0,0 +1,168 @@
+namespace AbcG = Alembic::AbcGeom;
+using namespace AbcG;
+
+
+int main(int argc, char **argv) {
+   
+    // Get arguments
+    CmdLine cmdLine;
+    std::string sfmDataFilename;
+    std::string sOutAlembic = "";
+
+    cmdLine.add(make_option('i', sfmDataFilename, "sfmdata"));
+    cmdLine.add(make_option('o', sOutAlembic, "outfile"));
+
+    try {
+        if (argc < 4) throw std::string("Invalid number of parameters in the command line.");
+        cmdLine.process(argc, argv);
+    } catch(const std::string &s) {
+        std::cout << "openMVG to alembic" << std::endl;
+        std::cout << "Usage: " << argv[0] << '\n'
+        << "[-i|--sfmdata filename, the SfM_Data file to convert]\n"
+        << "[-o|--outfile path]\n"
+        << std::endl;
+        std::cerr << s << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    //
+    SfM_Data sfm_data;
+    if (!Load(sfm_data, sfmDataFilename, ESfM_Data(ALL))) {
+        cout << "Error: The input project file \""<< sfmDataFilename << "\" cannot be read." << std::endl;
+        return EXIT_FAILURE;
+    }
+  
+    // Open Alembic archive
+    // TODO: decide wether we want to keep HDF5 or switch to Ogawa 
+    OArchive archive(Alembic::AbcCoreHDF5::WriteArchive(), sOutAlembic);
+    OObject topObj(archive, kTop);
+
+    //=================================================================================
+    // Export points
+    //=================================================================================
+
+    std::vector<V3f> positions;
+    positions.reserve(sfm_data.GetLandmarks().size());
+
+    // For all the 3d points in the hash_map
+    for(const auto lm : sfm_data.GetLandmarks()) {
+        const openMVG::Vec3 &pt = lm.second.X;
+        positions.emplace_back(pt[0], pt[1], pt[2]);
+    }
+
+    std::vector<Alembic::Util::uint64_t> ids(positions.size());
+    iota(begin(ids), end(ids), 0);
+
+    OPoints partsOut(topObj, "particleShape1");
+    OPointsSchema &pSchema = partsOut.getSchema();
+    OPointsSchema::Sample psamp( move(V3fArraySample (positions)), move(UInt64ArraySample ( ids )));
+    pSchema.set( psamp );
+
+    //=================================================================================
+    // Export cameras
+    //=================================================================================
+    for(const auto it: sfm_data.GetViews()) {
+        const View * view = it.second.get();
+        if (!sfm_data.IsPoseAndIntrinsicDefined(view))
+          continue;
+
+        // OpenMVG Camera
+        const openMVG::geometry::Pose3 pose = sfm_data.GetPoseOrDie(view);
+        auto iterIntrinsic = sfm_data.GetIntrinsics().find(view->id_intrinsic);
+        openMVG::cameras::Pinhole_Intrinsic *cam = static_cast<openMVG::cameras::Pinhole_Intrinsic*>(iterIntrinsic->second.get());
+        openMVG::Mat34 P = cam->get_projective_equivalent(pose);
+
+        // Extract internal matrix, rotation and translation
+        openMVG::Mat3 R, K;
```

Comment:
I tested with "pose.R()" but it is not in implemented in the version that I am using. Is this new function on the current develop branch ?